packages/lodestar/src/chain/chain.ts

```diff
@@ -222,4 +190,54 @@ export class BeaconChain extends EventEmitter implements IBeaconChain {
     //TODO: remove deposits with index <= newState.depositIndex
     await this.db.setMerkleTree(newState.depositIndex, merkleTree);
   }
+
+  private async checkGenesis(eth1Block: Block): Promise<void> {
+    this.logger.info(`Checking if block ${eth1Block.hash} will form valid genesis state`);
+    const deposits = await this.opPool.deposits.getAll();
+    const merkleTree = ProgressiveMerkleTree.empty(DEPOSIT_CONTRACT_TREE_DEPTH);
+    const depositsWithProof = deposits
+      .map((deposit, index) => {
+        merkleTree.add(index, hashTreeRoot(deposit.data, this.config.types.DepositData));
+        return deposit;
+      })
+      .map((deposit, index) => {
+        deposit.proof = merkleTree.getProof(index);
+        return deposit;
+      });
+    const genesisState = initializeBeaconStateFromEth1 (
+      this.config,
+      Buffer.from(eth1Block.hash.replace("0x", ""), "hex"),
+      eth1Block.timestamp,
+      depositsWithProof
+    );
+    if(!isValidGenesisState(this.config, genesisState)) {
+      this.logger.info(`Eth1 block ${eth1Block.hash} is NOT forming valid genesis state`);
+    }
+    this.logger.info(`Initializing beacon chain with eth1 block ${eth1Block.hash}`);
```

Comment:
Should I extract bottom part to public `initBeaconChain` method to get ahead of starting beaconchain from fake genesis state

Reply:
Just was about to leave a comment to do that xD